packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -88,10 +88,11 @@ export type FetcherResult =
   | string
   | { data: any };
 
-export type FetcherReturnType =
-  | Promise<FetcherResult>
-  | Observable<FetcherResult>
-  | AsyncIterable<FetcherResult>;
+type MaybePromise<T> = T | Promise<T>;
+
+export type FetcherReturnType = MaybePromise<
+  FetcherResult | Observable<FetcherResult> | AsyncIterable<FetcherResult>
```

Comment:
can we have a type for or even export this union, and then re-use it below on line 938?

Reply:
Like https://github.com/graphql/graphiql/pull/1739/commits/cd7061a7204e687dcdbd129725f196fb6f07ba5a?

Reply:
yes, ty!